.buildkite/autogenerate_pipeline.py

```diff
@@ -361,7 +393,18 @@ def generate_pipeline(config_file):
         "--test-description",
         metavar="JSON_FILE",
         help="The path to the JSON file containing the test" " description for the CI.",
-        default=f"{PARENT_DIR}/test_description.json",
+        default=f"{BUILDKITE_PATH}/test_description.json",
+    )
+    parser.add_argument(
+        "-p",
+        "--platform-allowlist",
+        metavar="PLATFORM_DOT_FILE",
+        help=(
+            "The path to the dot file containing the allowed platforms for the CI. "
```

Comment:
Let's document the format of this file somewhere (e.g. one architecture per line, and which architectures are supported in the first place). We'll also need to update the relevant parts of the README, which currently still states that CI will always run on x86_64 and ARM64 - here we should also mention the configurability now exposed via the .platforms file + its format.

Reply:
Current logic would obey lines read from `.platform` if present, should I add some constraint to enable `x86_64` and `arm64` on any circumstances 🤔

Reply:
> Current logic would obey lines read from `.platform` if present, should I add some constraint to enable `x86_64` and `arm64` on any circumstances 🤔

I think we should observe platforms provided by `.platform`, documentations are ready now